x-pack/plugins/ml/public/application/explorer/explorer_charts/explorer_charts_container.js

```diff
@@ -55,13 +55,25 @@ function getChartId(series) {
 
 // Wrapper for a single explorer chart
 function ExplorerChartContainer({ series, severity, tooManyBuckets, wrapLabel, mlUrlGenerator }) {
-  const redirectToSingleMetricViewer = useCallback(async () => {
-    const singleMetricViewerLink = await getExploreSeriesLink(mlUrlGenerator, series);
-    addItemToRecentlyAccessed('timeseriesexplorer', series.jobId, singleMetricViewerLink);
-
-    window.open(singleMetricViewerLink, '_blank');
+  const [explorerSeriesLink, setExplorerSeriesLink] = useState();
+
+  useEffect(() => {
+    let isCancelled = false;
+    const generateLink = async () => {
+      const singleMetricViewerLink = await getExploreSeriesLink(mlUrlGenerator, series);
+      if (!isCancelled) {
+        setExplorerSeriesLink(singleMetricViewerLink);
+      }
+    };
+    generateLink();
+    return () => {
+      isCancelled = true;
+    };
   }, [mlUrlGenerator]);
 
+  const addToRecentlyAccessed = useCallback(() => {
```

Comment:
Not related to the changes in this PR, but looks like the link added to the recently accessed list is wrong. For me it adds a link to (note `vlq/app/ml` is duplicated) e.g. 
```
http://localhost:5601/vlq/app/ml/vlq/app/ml/timeseriesexplorer?_g=(ml%3A(jobIds%3A!(gallery_mean_bytes_15m))%2CrefreshInterval%3A(display%3AOff%2Cpause%3A!t%2Cvalue%3A0)%2Ctime%3A(from%3A%272019-08-01T00%3A03%3A40.000Z%27%2Cmode%3Aabsolute%2Cto%3A%272020-12-16T11%3A33%3A45.000Z%27))&_a=(timeseriesexplorer%3A(mlTimeSeriesExplorer%3A(detectorIndex%3A0%2Centities%3A(method%3APOST))%2Cquery%3A(query_string%3A(query_string%3A(analyze_wildcard%3A!t%2Cquery%3A%27*%27)))))
```

Reply:
Fixed here 7f82298 (#86182). Seems like this is also an issue on 7.10.2 so I'll be making a backport to 7.10 as well.